test/common/upstream/load_balancer_fuzz_base.cc

```diff
@@ -0,0 +1,130 @@
+// setting the mock priority set to have certain values, and all load balancers interface with host
+// sets and their health statuses, this action maps to all load balancers.
+void LoadBalancerFuzzBase::updateHealthFlagsForAHostSet(uint64_t host_index,
+                                                        uint32_t num_healthy_hosts,
+                                                        uint32_t num_degraded_hosts,
+                                                        uint32_t num_excluded_hosts) {
+  uint8_t index_of_host_set = host_index % num_host_sets_;
+  ENVOY_LOG_MISC(trace, "Updating health flags for host set: {}", index_of_host_set);
+  MockHostSet& host_set = *priority_set_.getMockHostSet(index_of_host_set);
+  // This downcast will not overflow because size is capped by port numbers
+  uint32_t host_set_size = host_set.hosts_.size();
+  host_set.healthy_hosts_.clear();
+  host_set.degraded_hosts_.clear();
+  host_set.excluded_hosts_.clear();
+  // The vector constructed here represents remaining indexes, an index will randomly get chosen
+  // from here every time and removed. Thus, this represents the remaining indexes. When this goes
+  // to zero, there will be no more indexes left to place into healthy, degraded, and excluded
+  // hosts.
+  ENVOY_LOG_MISC(trace, "Host set size: {}", host_set_size);
+  std::vector<uint8_t> index_vector;
+  index_vector.reserve(host_set_size);
+  for (uint8_t i = 0; i < host_set_size; i++) {
+    index_vector.push_back(i);
+  }
+
+  for (uint32_t i = 0; i < num_healthy_hosts && index_vector.size() != 0; i++) {
+    uint64_t index_of_index_vector = random_.random() % index_vector.size();
+    uint64_t index = index_vector.at(index_of_index_vector);
+    ENVOY_LOG_MISC(trace, "Added host {} to healthy hosts for host set {}", index,
+                   index_of_host_set);
+    host_set.healthy_hosts_.push_back(host_set.hosts_[index]);
+    index_vector.erase(index_vector.begin() + index_of_index_vector);
+  }
+
+  for (uint32_t i = 0; i < num_degraded_hosts && index_vector.size() != 0; i++) {
+    uint64_t index_of_index_vector = random_.random() % index_vector.size();
+    uint64_t index = index_vector.at(index_of_index_vector);
+    ENVOY_LOG_MISC(trace, "Added host {} to degraded hosts for host set {}", index,
+                   index_of_host_set);
+    host_set.degraded_hosts_.push_back(host_set.hosts_[index]);
+    index_vector.erase(index_vector.begin() + index_of_index_vector);
+  }
+
+  for (uint32_t i = 0; i < num_excluded_hosts && index_vector.size() != 0; i++) {
+    uint64_t index_of_index_vector = random_.random() % index_vector.size();
+    uint64_t index = index_vector.at(index_of_index_vector);
+    ENVOY_LOG_MISC(trace, "Added host {} to excluded hosts for host set {}", index,
+                   index_of_host_set);
+    host_set.excluded_hosts_.push_back(host_set.hosts_[index]);
+    index_vector.erase(index_vector.begin() + index_of_index_vector);
+  }
```

Comment:
I'm wondering whether you can just take the hosts indices vector, [0, host.set_hosts_.size()), shuffle it (using the prng value), and then assign the first 'num_healthy_hosts' indices to be the healthy ones, the next 'num_degraded_hosts' to be the degraded ones, and the rest to be excluded.
I believe this can be done if num_healthy+num_degraded+num_excluded >= hosts.set_hosts_.size(), and if the order of assignment doesn't matter.
This should avoid erasing elements from the middle of a vector.